File: packages/core/src/tool.ts
import { z } from "zod";

/**
 * Tool input/output schema placeholder; concrete schemas in @agent-framework/tools.
 */
export const ToolDefinitionSchema = z.object({
  name: z.string(),
  description: z.string().optional(),
});
export type ToolDefinition = z.infer<typeof ToolDefinitionSchema>;

export interface ToolResult<T = unknown> {
  success: boolean;
  data?: T;
  error?: string;
}

File: python/src/agent_framework/tool.py
"""Tool definition and result types."""

from pydantic import BaseModel
from typing import Any


class ToolDefinition(BaseModel):
    """Metadata for a tool."""

    name: str
    description: str | None = None


class ToolResult(BaseModel):
    """Result of a tool invocation."""

    success: bool
    data: Any = None
    error: str | None = None
